lib/msal-angular/docs/v2-docs/ssosilent.md

```diff
@@ -20,36 +23,55 @@ If ssoSilent() fails, we recommend handling this error by logging in interactive
 
 ```js
 import { Component, OnInit } from '@angular/core';
-import { MsalService } from '@azure/msal-angular';
-import { SilentRequest, SsoSilentRequest } from '@azure/msal-browser';
+import { MsalService, MsalBroadcastService } from '@azure/msal-angular';
+import { SilentRequest, SsoSilentRequest, InteractionStatus, AuthenticationResult } from '@azure/msal-browser';
+import { Subject } from 'rxjs';
+import { filter, takeUntil } from 'rxjs/operators';
 
 @Component({
-  selector: 'app-root',
-  templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css']
+    selector: 'app-root',
+    templateUrl: './app.component.html',
+    styleUrls: ['./app.component.css']
 })
 export class AppComponent implements OnInit {
+    private readonly _destroying$ = new Subject<void>();
 
-  constructor(
-    private authService: MsalService,
-  ) {}
+    constructor(
+        private authService: MsalService,
+        private msalBroadcastService: MsalBroadcastService
+    ) {}
+
+    ngOnInit(): void {
+        const silentRequest: SsoSilentRequest = {
+            scopes: ["User.Read"],
+            loginHint: "user@contoso.com"
+        }
 
-  ngOnInit(): void {
-    const silentRequest: SsoSilentRequest = {
-      scopes: ["User.Read"],
-      loginHint: "user@contoso.com"
+        // Wait until pending interaction is complete
+        this.msalBroadcastService.inProgress$
+            .pipe(
+                filter((status: InteractionStatus) => status === InteractionStatus.None),
+                takeUntil(this._destroying$)
+            )
+            .subscribe(() => {
+                // Check current accounts
+                if (this.authService.instance.getAllAccounts().length < 1) {
```

Comment:
This is really only needed when using redirect, although does help prevent brief flash when using popup.

Reply:
Could add this to docs or comments